Source/msg.cpp

```diff
@@ -1537,9 +1548,9 @@ DWORD OnPlayerJoinLevel(const TCmd *pCmd, int pnum)
 		EventPlrMsg(fmt::format(_("Player '{:s}' (level {:d}) just joined the game"), player._pName, player._pLevel).c_str());
 	}
 
-	if (player.plractive && MyPlayerId != pnum) {
-		player.position.tile = { p->x, p->y };
-		player.plrlevel = p->wParam1;
+	if (player.plractive && MyPlayerId != pnum && message.wParam1 < MAXCHARLEVEL && InDungeonBounds(position)) {

```

Comment:
This could leave the game in a inconsistent state cause the activated the player but didn't add assign him to any level. (For example what content has `player.plrlevel` and `player.position.tile`. Can they could contain uninitialized variables with negative values? If yes what would happen later int the game logic?

Reply:
Moved the check up so that it bails before announcing the player.